### PR TITLE
Announce CNES/`demcompare` collaboration and update lead team background

### PR DESCRIPTION
**Pull request by rhugonnet**
This PR updates the documentation to announce the collaboration and lead team change with `demcompare`.

A preview of the documentation generated on this branch can be found here: https://xdem-rhugonnet.readthedocs.io/en/doc_demcompare/

Specifically, this PR adds:
- An announcement about the xDEM-demcompare collaboration **on the "Where to start?" landing page**,
- A short description of who develops/uses xDEM **on the "About xDEM" page**, mentioning the joint lead from CNES + researchers,
- A longer description and funding acknowledgments about the current leading team **on the "Background" page (bottom)**.
- Adds logos of CNES, SNSF and NASA on the last two pages, being the main funding sources behind xDEM!

## TO-DO

- [x] Add all grant numbers to the funding sections,
- [x] Add light+dark logo versions to support both themes (NASA is the same; SNSF is available on their website, CNES not sure).

### CODE REVIEW

**Comment by rhugonnet** [comment]
@duboise-cnes @adebardo @vschaffn This PR is ready for your review! There is a link in the PR description above to visualize the changes from this PR in the doc.
Feel free to propose edits or add new statements. I kept it short everywhere, but we could go into more details about the fusion plan, or CNES expertise in the "Background" page for instance. Just tell me what you think :wink:.

**(In addition to this PR) New documentation**
Additionally, if you want, you can browse the new documentation (merged a couple days ago in #502; a big endeavor that started nearly 6 months ago) and note any remark you have! :slightly_smiling_face:
The link above for this PR is up-to-date with the new doc so that works, or it is also available on the `latest` branch online: https://xdem.readthedocs.io/en/latest/ (the `latest` branch of the public documentation is not the default so not too visible yet, the default `stable` is still the old doc).
I merged it separately after Erik and Amaury's reviews so that this PR would be easier for you to browse, and because the tracked changes of the massive #502 PR weren't very useful for reviewing anyway (so much changed, and it's always good to re-read surrounding stuff when updating the doc). 
So the best way to review is rather to browse the doc itself, page by page!

**Short summary of changes to help you get an overview:** The new doc separates scientific background content and code examples (moving and adding scientific content into guide pages in "Resources"), adds better links and plots to all feature pages, and adds some auxiliary pages for diverse uses based on user feedback (how to cite, method overview, cheatsheet for helping selecting a method, configuration, ecosystem, release notes, etc). It also improves the API listing, and adds a lot of streamlining/improvements for consistency, including for example a "Quick use" section at the start of each feature page, and tables summarizing available methods on most of those pages too.

And, with the new doc nearly finalized, we had in mind to follow up with the release a `0.1` (maybe once `BlockwiseCoreg` is fixed!).

File: doc/source/background.md [hunk 1]
```diff
@@ -2,29 +2,97 @@
...
+Other volunteer contributors span diverse scientific backgrounds in industry or research. We welcome
+any new contributors! See how to contribute on [the dedicated page of our repository](https://github.com/GlacioHack/xdem/blob/main/CONTRIBUTING.md).
+
+## Funding acknowledgments
+
+Members of the lead development team acknowledge funding from:
+- SNSF grant no. 184634, a MeteoSwiss [GCOS](https://gcos.wmo.int/en/home) project on elevation data analysis for glaciology,
+- NASA award 80NSSC22K1094, an [STV](https://science.nasa.gov/earth-science/decadal-surveys/decadal-stv/) project on the fusion of elevation data,
+- NASA award 80NSSC23K0192, an [ICESat-2](https://icesat-2.gsfc.nasa.gov/) project on the processing of elevation data in the cloud,
+- CNES (French Space Agency): ?.
```
Comment (duboise-cnes) [inline doc/source/background.md:45]: do you need a number ? I don't know which award would be "public" 
here. Maybe: study on merging demcompare and xdem and development of demcompare functions into xdem
or better written ;)

**Comment by duboise-cnes** [comment]
Thanks @rhugonnet . Perfect for me. Thanks for the add-on.

File: doc/source/background.md (hunk 1 above)
Comment (rhugonnet) [reply doc/source/background.md]: No number necessary, I added the sentence! :)

**Comment by rhugonnet** [comment]
Perfect! Comments accounted for. 
Merging this PR now for the addition, but don't hesitate to send comments on the new doc separately, if you have any!

**Pull request merged**